src/main/java/io/mosip/mimoto/service/impl/WalletCredentialServiceImpl.java

```
@@ -207,4 +207,22 @@ private VerifiableCredential saveCredential(String walletId, String encryptedCre
 
         return walletCredentialsRepository.save(verifiableCredential);
     }
+
+    @Override
+    public boolean deleteCredential(String credentialId, String walletId) {
+        log.info("Deleting credential with ID: {} for wallet: {}", credentialId, walletId);
+
+        // Find the credential by ID and wallet ID to ensure it belongs to the specified wallet
+        Optional<VerifiableCredential> credential = walletCredentialsRepository.findById(credentialId);
+
+        if (credential.isPresent() && credential.get().getWalletId().equals(walletId)) {
+            // Delete the credential
+            walletCredentialsRepository.deleteById(credentialId);
+            log.info("Successfully deleted credential with ID: {}", credentialId);
+            return true;
+        } else {
+            log.warn("Credential with ID: {} not found or does not belong to wallet: {}", credentialId, walletId);
+            return false;
```

Instead of returning false please throw the error so that we can return that in the response to the caller.